app/services/Agents.scala

```diff
@@ -32,23 +31,23 @@ class Agents @Inject() (
 )(implicit exec: ExecutionContext)
     extends Logging {
 
-  lazy implicit val conf = amiableConfigProvider.conf
+  lazy implicit val conf: AMIableConfig = amiableConfigProvider.conf
   val refreshInterval = 5.minutes
 
-  private val amisAgent: Agent[Set[AMI]] = Agent(Set.empty)
-  private val ssasAgent: Agent[Set[SSAA]] = Agent(Set.empty)
-  private val oldProdInstanceCountAgent: Agent[Option[Int]] = Agent(None)
+  private val amisAgent: AtomicReference[Set[AMI]] = AtomicReference(Set.empty)
+  private val ssasAgent: AtomicReference[Set[SSAA]] = AtomicReference(Set.empty)
+  private val oldProdInstanceCountAgent: AtomicReference[Option[Int]] = AtomicReference(None)
   private val oldProdInstanceCountHistoryAgent
-      : Agent[List[(DateTime, Double)]] = Agent(Nil)
-  private val amisAgePercentilesAgent: Agent[Option[Percentiles]] = Agent(None)
+      : AtomicReference[List[(DateTime, Double)]] = AtomicReference(Nil)
+  private val amisAgePercentilesAgent: AtomicReference[Option[Percentiles]] = AtomicReference(None)
   private val amisAgePercentile25thHistoryAgent
-      : Agent[List[(DateTime, Double)]] = Agent(Nil)
+      : AtomicReference[List[(DateTime, Double)]] = AtomicReference(Nil)
   private val amisAgePercentile50thHistoryAgent
-      : Agent[List[(DateTime, Double)]] = Agent(Nil)
+      : AtomicReference[List[(DateTime, Double)]] = AtomicReference(Nil)
   private val amisAgePercentile75thHistoryAgent
-      : Agent[List[(DateTime, Double)]] = Agent(Nil)
+      : AtomicReference[List[(DateTime, Double)]] = AtomicReference(Nil)
   private val oldInstanceCountByAccountAgent
-      : Agent[List[OldInstanceAccountHistory]] = Agent(Nil)
+      : AtomicReference[List[OldInstanceAccountHistory]] = AtomicReference(Nil)
```

Comment:
The `Agent` class no longer exists in Pekko (it was removed in Akka 2.5). As I understand this class was meant to be used to ensure whatever data the agent wraps is accessed in a thread-safe & lock-free manner.

Looking at how Amiable uses `Agent` it seems it doesn't actually need to be thread safe, Only one thread writes to each Agent and it overrides the value each time.

Reply:
We use `scheduleWithFixedDelay` to start the thread that writes to these `Agents`/`AtomicReferences`, as I understand the Pekko scheduler wont start a new thread until the old one has finished, so it _should_ only be getting written to by one thread at a time.

I guess the risk would be that the writer thread doesn't write everything in one go. For example if the writer thread pushes one value at a time to an array and that array is accessed by a reader whilst the writer is still working, the reader could get incomplete data, however in this case we only use `.set` operations replacing one complete array with another complete array.